numbagg/moving.py

```diff
@@ -51,42 +43,44 @@ def move_exp_nanmean(a, alpha, out):
 
 
 @ndmoving(
-    [
-        (float64[:], int64, float64[:]),
-        (float32[:], int64, float32[:]),
-        (float64[:], int32, float64[:]),
-        (float32[:], int32, float32[:]),
-    ]
+    [(float32[:], int64, int64, float32[:]), (float64[:], int64, int64, float64[:])]
 )
-def move_nanmean(a, window, out):
+def move_mean(a, window, min_count, out):
 
     asum = 0.0
     count = 0
-    for i in range(window - 1):
+
+    for i in range(min_count - 1):
         ai = a[i]
-        if ai == ai:
+        if not np.isnan(ai):
             asum += ai
             count += 1
         out[i] = np.nan
-    i = window - 1
-    ai = a[i]
-    if ai == ai:
-        asum += ai
-        count += 1
-    if count > 0:
-        out[i] = asum / count
-    else:
-        out[i] = np.nan
-    for i in range(window, len(a)):
+
+    for i in range(min_count - 1, window):
         ai = a[i]
-        if ai == ai:
+        if not np.isnan(ai):
             asum += ai
             count += 1
+        out[i] = asum / count if count >= min_count else np.nan
+
+    count_inv = 1 / count if count >= min_count else np.nan
```

Comment:
Out of interest, is there an easy explanation why we calc `1 / count` and multiply that by `asum`, rather than calcing `asum / count`?

Reply:
It has approximately the same performance, but otherwise, Numba will issue a warning for division by 0 (if `count` is 0).

I should really add a comment, though...

Reply:
Makes sense, thx